rbd/encryption_opt_luks.go

```diff
@@ -0,0 +1,34 @@
+//go:build !octopus && !pacific && !quincy && ceph_preview
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// /* force XSI-complaint strerror_r() */
+// #define _POSIX_C_SOURCE 200112L
+// #undef _GNU_SOURCE
+// #include <stdlib.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+// EncryptionOptionsLUKS generic options for either LUKS v1 or v2. May only be
+// used for opening existing images - not valid for the EncryptionFormat call.
+type EncryptionOptionsLUKS struct {
```

Comment:
Nit: I'd place both this and `EncryptionLoad2` implementation in `encryption.go`, so that that all LUKS structs and related APIs are together.

Reply:
Thanks for the suggestion, but that would violate the project's workflow and ability to build on older ceph releases using go build tags. Because of user demand we older ceph versions so merging this into encryption.go would break the ability to build against things prior to reef (Go eschews IFDEFs, it accomplishes similar stuff with build tags which than implies adding additional files).

Reply:
I see, thanks for the explanation.   Since both `encryption_opt_luks.go` and `encryption_load2.go` have the same go build tag, perhaps merge them into a single `encryption_reef.go` file then?

Feel free to just resolve this thread with no further comment if the way it is is preferred ;)